### PR TITLE
ETSWORK-9564 - Add and configure react-i18next to support internationalization

### PR DESCRIPTION
**Pull request by coreyjv**
`react-18next` was the library chosen by Workgrid Engineering to use in our client applications. This initial configuration assumes we'll be able to fetch translations over the network at some point in the future or if we deploy as as webapp. In its present form for mobile the translations will be bundled with the application. Even though we're still using the blank Ionic template I've provided an example usage of the translation hook to demonstrate how it works. This is only temporary and it will be removed once we begin actual development of features.

### CODE REVIEW

File: src/pages/Home.tsx [hunk 1]
```diff
@@ -14,7 +17,7 @@ const Home: React.FC = () => {
       <IonContent fullscreen>
         <IonHeader collapse="condense">
           <IonToolbar>
-            <IonTitle size="large">Blank</IonTitle>
+            <IonTitle size="large">{t('title')}</IonTitle>
```
Comment (Kikketer) [inline src/pages/Home.tsx:20]: We've decided as a styling issue to have the namespace included in these strings:
```
t('home:title')
```
It just makes it slightly easier to read and understand/debug when you are missing some languages.

Comment (coreyjv) [reply src/pages/Home.tsx:20]: @Kikketer --

In the example in the code the namespace was provided to the hook so one doesn't need to include it in the `t` function. This was just an example to illustrate that it was working as intended and will likely be removed as we replace the template views with actual views. 

Can I propose that we merge this change now and discuss this best practice and if/how we'd go about enforcing it? 

Thanks!

Comment (coreyjv) [reply src/pages/Home.tsx:20]: I spent a bit of time to see if there was any way we could automate a check like this if we chose to adopt it. I couldn't find any ESLint rules that could enforce this so any enforcement would have to be manual (unless we built a custom rule).

Comment (Kikketer) [reply src/pages/Home.tsx:20]: Yeah I realize it's impossible to enforce except by either having everyone on the same page and trusting it or manually checking.  Some may be missed but if we can start now and get everyone on the same page of including the namespace we can get started in the right/consistent spot.

Juggling i18n is already a little painful (where is this text? is this text being used? why isn't it showing the text even if I see it in X? etc).  If we specifically have the namespace in the specific translation, it'll help at least a little bit.

File: src/i18n.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    interpolation: {
+      escapeValue: false // not needed for react as it escapes by default
+    },
+    backend: {
+      backends: [
+        I18NextLocalStorageBackend, // primary
+        I18NextHttpBackend // This will allow us to centrally host translations that can be pulled by any app
+      ],
+      backendOptions: [
+        {},
```
Comment (Kikketer) [inline src/i18n.ts:25]: Is this doing something fancy?

**Review by Kikketer — APPROVED** [review]

File: src/pages/Home.tsx (hunk 1 above)
Comment (coreyjv) [reply src/pages/Home.tsx:20]: Let's discuss this internally and figure out the problem, our options, and how we'd like to proceed.

**Review by louise-hayes — APPROVED** [review]

File: src/i18n.ts (hunk 2 above)
Comment (coreyjv) [reply src/i18n.ts:25]: Not really. The way `i18next` works is that if you add a back-end and you want to provide options to it you need to provide those options in the same _array position_ in `backendOptions`. So the way this reads is that no options are provided to the `I18NextLocalStorageBackend` but we are providing options to the `I18NextHttpBackend`. I hope this adds clarification.

**Pull request merged**